engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -263,22 +276,60 @@ def has_no_regcode?(auth_header)
 
         protected
 
+        # rubocop:disable Metrics/PerceivedComplexity
+        # rubocop:disable Metrics/CyclomaticComplexity
+        # rubocop:disable Metrics/AbcSize
         def scc_activate_product
```

Comment:
This method needs refactoring to be easier to understand. This can be refactored in another pr, but it's important that this workflow is readable. How about a structure like this: 

```
# Proxying BYOS product activations to SCC
def scc_activate_product
  if byos_activation? 
    SccProxy.announce_system_scc unless @system.scc_id 
    SccProxy.scc_activate_product
  end
end
```

Reply:
I agree on the refactoring, yes, not only that but more stuff can be refactored on that method